File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    participants = {}

    for p in participant:
        if p in participants.keys():
            participants[p] += 1
            
        else:
            participants[p] = 1
            
    for c in completion:
        participants[c] -= 1
        
        if participants[c] == 0:
            del participants[c]
        
    return list(participants.keys())[0]
    

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {    
    unordered_map<string, int> um;
    
    for(const string& p: participant) {
        um[p] += 1;
    }
    
    for(const string& c: completion) {
        um[c] -= 1;
    }

    auto answer = find_if(um.begin(), um.end(), [](const auto& it) {        return it.second != 0;
    });
    
    return answer -> first;
}